src/providers/token-provider.ts

```diff
@@ -1,4 +1,5 @@
 import { Interface } from '@ethersproject/abi';
+import { parseBytes32String } from '@ethersproject/strings';
```

Comment:
What does this func do? strip the tailing 0s off of the bytes string or try converting it into ASCII?

Reply:
this is the JS function:
```
function parseBytes32String(bytes) {
    var data = (0, bytes_1.arrayify)(bytes);
    // Must be 32 bytes with a null-termination
    if (data.length !== 32) {
        throw new Error("invalid bytes32 - not 32 bytes long");
    }
    if (data[31] !== 0) {
        throw new Error("invalid bytes32 string - no null terminator");
    }
    // Find the null termination
    var length = 31;
    while (data[length - 1] === 0) {
        length--;
    }
    // Determine the string value
    return (0, utf8_1.toUtf8String)(data.slice(0, length));
}
```

it appears like the `toUtf8String` does the conversion

Reply:
bcc from the library:

```
export function parseBytes32String(bytes: BytesLike): string {
    const data = arrayify(bytes);

    // Must be 32 bytes with a null-termination
    if (data.length !== 32) { throw new Error("invalid bytes32 - not 32 bytes long"); }
    if (data[31] !== 0) { throw new Error("invalid bytes32 string - no null terminator"); }

    // Find the null termination
    let length = 31;
    while (data[length - 1] === 0) { length--; }

    // Determine the string value
    return toUtf8String(data.slice(0, length));
}
```

so 1) truncating trailing 0s 2) convert to utf8.